### ISSUE TITLE
Does not work on Safari for MacOS

### ISSUE THREAD

**Issue by CollinsVakayil** [GoogleChromeLabs/link-to-text-fragment#69]

**Comment by tomayac** [comment]
I can reproduce, but have no idea what has changed. It was working fine at the time the extension was released. I need to investigate, but can’t really commit to a time this will happen. If you have cycles, happy for your help!

**Comment by CollinsVakayil** [comment]
> I can reproduce, but have no idea what has changed. It was working fine at the time the extension was released. I need to investigate, but can’t really commit to a time this will happen. If you have cycles, happy for your help!

I'm happy to help any way I can, unfortunately I don't know how to code. So forgive my ignorance on terminology when I ask what are cycles and how do get them to you.

**Comment by ahmetasabanci** [comment]
Also having the same issue.

@tomayac If there's anything I can do to help you investigate, just let me know.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
@tomayac, do you know how to automate enablement of that feature via the command-line, or how I might ascertain how to?

**Comment by tomayac** [comment]
Changed the title to reflect the related problem reported in the closed #62.

**Comment by tomayac** [comment]
> @tomayac, do you know how to automate enablement of [that feature](https://github.com/GoogleChromeLabs/link-to-text-fragment/issues/69#issuecomment-1185614230) via the command-line, or how I might ascertain how to?

I don't know, sorry.

**Comment by jasmas** [comment]
I get an error in the javascript console in Safari that reads:
Error: Unchecked runtime.lastError: This extension already contains a context menu item with identifier 'copy-link'.

Not sure if that's related or helpful

**Comment by JayHoltslander** [comment]
I was so excited for this upon discovering the existence of text fragments this morning. Sadly the Safari version of the plugin is not working for me on my 5 year old Mac running Mac OS 11.7.1 (The highest it can go.) Clipboard is not populated with a link and instead remains full of previously copied text. I'm sad because in my day to day surfing I prefer Safari over Chrome.

**Comment by rokejulianlockhart** [comment]
@JayHoltslander, have you tried Linux on your Mac? I don't think that asking for support for 5 year old Safari is very reasonable.

**Comment by rokejulianlockhart** [comment]
Indeed, that sort of age for most computers isn't problematic, but you're running an unsupported OS. That universally is.

**Comment by jdelStrother** [comment]
I poked at this for a bit, though without much success.

The first problem is that [injectContentScripts](https://github.com/GoogleChromeLabs/link-to-text-fragment/blob/06a0a7a1215435edde5b89d7acdbc10d67174868/background_script.js#L30) pings the tab to see if the content script has been injected, but `browser.tabs.sendMessage` never completes - it just seems to behave like a black hole.

I can force the injection with something like this - 

```diff
--- a/LinkToText Extension/Resources/background_script.js
+++ b/LinkToText Extension/Resources/background_script.js

@@ -26,26 +26,33 @@
       console.log(...args);
     }
   };

+let firstTime = true;
   const injectContentScripts = async (contentScriptNames) => {
     // If there's a reply, the content script already was injected.
     try {
+        if (firstTime) {
+          firstTime = false
+          throw new Error('force injection')
+        }
      return await sendMessageToPage('ping');
     } catch (err) {
```

after which sendMessageToPage behaves as expected, and the content script generates a linked url.

Unfortunately, even then copying doesn't work. It tries to use the [fallback copy method](https://github.com/GoogleChromeLabs/link-to-text-fragment/blob/06a0a7a1215435edde5b89d7acdbc10d67174868/content_script.js#L137) but nothing happens (maybe because Safari will only copy-to-clipboard on a user event and it's not seeing this as a user event?).

**Comment by tomayac** [comment]
It’s surprisingly frustrating, especially since it worked fine when I released it initially on the App Store. I personally can’t justify the time investment at the moment. Thanks for investigating, though!
